caffe2/operators/layer_norm_op.cc

```diff
@@ -196,12 +196,32 @@ void layer_norm_c10(c10::Stack* stack, c10::KernelCache* cache_) { // TODO Pass
   c10::ArrayRef<c10::IValue> inputs = torch::jit::peekSlice(*stack, 0, 3, 6);
   c10::ArrayRef<c10::IValue> outputs = torch::jit::peekSlice(*stack, 3, 3, 6);
```

Comment:
I'm still confused on what are the semantics of passing outputs. They are marked as optional, so when called from PT frontend they'd usually be just empty IValue, right?

Reply:
Yes, for caffe2, outputs are added at the end of the argument list as optional arguments, i.e. when called from a frontend that doesn't care about them (say eager mode), they will be an empty IValue. They're also defaulted to None, so user code in python doesn't have to specify them.

https://github.com/pytorch/pytorch/pull/16548/files adds comments that describe the stack behavior for caffe2 ops.